### PR TITLE
add Gluon PReLU activation layer

### PR DESCRIPTION
**Pull request by joeddav**
## Description ##
Adds a PReLU layer for Gluon per #8898. 

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] For user-facing API changes, API doc string has been updated. For new C++ functions in header files, their functionalities and arguments are well-documented. 
- [x] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

### CODE REVIEW

File: python/mxnet/gluon/nn/basic_layers.py [hunk 1]
```diff
@@ -416,6 +416,48 @@ def __repr__(self):
                         alpha=self._alpha)
 
 
+class PReLU(HybridBlock):
```
Comment (piiswrong) [inline python/mxnet/gluon/nn/basic_layers.py]: what's the difference between this and leakyrelu? Should it be an option of leakyrelu?

Comment (joeddav) [reply python/mxnet/gluon/nn/basic_layers.py]: The difference is leaky relu takes a constant, whereas prelu creates a learnable parameter.

Comment (piiswrong) [reply python/mxnet/gluon/nn/basic_layers.py]: Is RReLU a standard name? Could you add reference to the paper?

Comment (piiswrong) [reply python/mxnet/gluon/nn/basic_layers.py]: I remember there are a bunch of relus with similar names.

Comment (joeddav) [reply python/mxnet/gluon/nn/basic_layers.py]: Added. Yeah there are a bunch, but I think PReLU is a fairly common one. I know it's built in to PyTorch, Keras for example. I'm not familiar with RReLU.

**Comment by piiswrong** [comment]
Could you add unittest and rebase to master? see tests/python/unittests/test_gluon_xxx.py for example

**Comment by bradcar** [comment]
Open question: Since Gluon LeakyReLU is implemented with 
<pre> return F.LeakyReLU(x, act_type='leaky', slope=self._alpha, name='fwd')</pre>
and since there is a act_type='prelu' in  F.LeakyReLU, why not also implement Gluon PReLU with 
<pre> return F.LeakyReLU(x, act_type='prelu', ...etc</pre>
What would be more efficient?

**Comment by joeddav** [comment]
Good call @bradcar, I wasn't aware of that parameter. I'll change the implementation to that approach instead, and then add tests and rebase, @piiswrong

**Comment by bradcar** [comment]
any update on using  ...return F.LeakyReLU(x, act_type='prelu', ...etc?

**Comment by piiswrong** [comment]
@yzhliu @szha 
The original author seems to have disappeared. Can one of the committers take over?
Please address the comments and add tests.

**Comment by szha** [comment]
working on it in the above PR

**Pull request closed without merging**